src/extended_json.ts

```diff
@@ -70,17 +72,23 @@ const keysToCodecs = {
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function deserializeValue(value: any, options: EJSONOptions = {}) {
   if (typeof value === 'number') {
+    const in32BitRange = value <= BSON_INT32_MAX && value >= BSON_INT32_MIN;
+    const in64BitRange = value <= BSON_INT64_MAX && value >= BSON_INT64_MIN;
+
     if (options.relaxed || options.legacy) {
-      return value;
+      return Number.isInteger(value) && options.useBigInt64 ? BigInt(value) : value;
```

Comment:
I don't think relaxed: true is supposed to have any bigint logic, looking at the examples we should keep returning value here